src/Microsoft.Data.SqlClient/netcore/src/Common/src/Microsoft/Data/Common/DbConnectionOptions.Common.cs

```diff
@@ -166,7 +170,7 @@ private static void DebugTraceKeyValuePair(string keyname, string keyvalue, Dict
                 Debug.Assert(string.Equals(keyname, keyname?.ToLower(), StringComparison.InvariantCulture), "missing ToLower");
                 string realkeyname = ((null != synonyms) ? (string)synonyms[keyname] : keyname);
 
-                if ((KEY.Password != realkeyname) && (SYNONYM.Pwd != realkeyname))
+                if ((KEY.Password != realkeyname) && (SYNONYM.Pwd != realkeyname) && (KEY.ClientKeyPassword != realkeyname))
```

Comment:
Why is this needed?

Reply:
Well i guess for security reasons you want to remove plain-text password from memory into SecureString
that would mean you want to remove all kind of passwords from connection string.